subsys/nrf_security/Kconfig.tls

```diff
@@ -256,6 +260,22 @@ config MBEDTLS_SSL_KEEP_PEER_CERTIFICATE
 	  reduces RAM usage.
 	  Corresponds to MBEDTLS_SSL_KEEP_PEER_CERTIFICATE in mbed TLS config file.
 
+config MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE
```

Comment:
I see [this defaults to on](https://github.com/zephyrproject-rtos/zephyr/blob/86b666548b97f0fa5f4ec05e9b6f92a2bfe5a4ed/modules/mbedtls/configs/config-tls-generic.h#L63) in Zephyr. Not sure what is the right thing to do?

Reply:
If i'm not mistaken, this is a workaround for TLS1.2 middleboxes. Which adds a certain latency to the handshake. At first I figured there was no use in defaulting it to `y`. But apparently the drawback isn't big and it doesn't affect anything that much, so i'll just put it back.

Reply:
@valeriosetti Feel free to chime in if you have anything more to add.